src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/mailActionFields.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import SelectMembers from 'app/components/selectMembers';
+import SelectControl from 'app/components/forms/selectControl';
+import {Organization, Project} from 'app/types';
+import {IssueAlertRuleAction} from 'app/types/alerts';
+import {PanelItem} from 'app/components/panels';
+import space from 'app/styles/space';
+import withProjects from 'app/utils/withProjects';
+import withProject from 'app/utils/withProject';
+import withOrganization from 'app/utils/withOrganization';
+
+type Props = {
+  project: Project;
+  organization: Organization;
+  projects: Project[];
+  disabled: boolean;
+  loading: boolean;
+  action: IssueAlertRuleAction;
+  onChange: (action: IssueAlertRuleAction) => void;
+};
+
+export enum MailActionTargetType {
+  IssueOwners = 'IssueOwners',
+  Team = 'Team',
+  Member = 'Member',
+}
+
+class MailActionFields extends React.PureComponent<Props> {
+  handleChange = (attribute: 'targetType' | 'targetIdentifier', e: HTMLInputElement) => {
+    const {onChange, action} = this.props;
+    if (e.value === action[attribute]) {
+      return;
+    }
+    const newAction = {...action};
+    newAction[attribute] = `${e.value}`;
+    /**
+     * TargetIdentifiers between the targetTypes are not unique, and may wrongly map to something that has not been
+     * selected. E.g. A member and project can both have the `targetIdentifier`, `'2'`. Hence we clear the identifier.
+     **/
+    if (attribute === 'targetType') {
+      newAction.targetIdentifier = '';
+    }
+    onChange(newAction);
+    /**
+     * Force update is needed because parent component mutates its props with the new values that have changed.
+     * Effectively, this causes `action`, a nested prop from the parent, to be referentially the same even if a new
+     * `action` is passed to the change handler.
+     **/
+    this.forceUpdate();
+  };
+
+  handleChangeActorType = (e: HTMLInputElement) => {
```

Comment:
I don't  think this type is correct, `<SelectControl>` calls onChange with the option object that was selected, (e.g {value, label}).

Reply:
Done. Thanks!